packages/cli/api-importers/v2-importer-commons/src/AbstractConverter.ts

```diff
@@ -90,6 +92,66 @@ export abstract class AbstractConverter<Context extends AbstractConverterContext
 
+    protected async resolveExternalRefs({ spec, context }: { spec: unknown; context: Context }): Promise<unknown> {
+        const queue = [spec];
+
+        while (queue.length > 0) {
+            const current = queue.shift();
+            if (current == null) {
+                continue;
+            }
+
+            if (Array.isArray(current)) {
+                for (let i = 0; i < current.length; i++) {
+                    let resolvedRefVal = current[i];
+                    if (this.context.isReferenceObject(current[i])) {
+                        // repeatedly resolve nested refs until we get a definition to check whether it's external
+                        while (this.context.isReferenceObject(resolvedRefVal)) {
+                            const externalRef = this.context.isExternalReference(resolvedRefVal.$ref);
+                            const nextResolvedRef = await context.resolveReference({ $ref: resolvedRefVal.$ref });
+                            if (nextResolvedRef.resolved) {
+                                resolvedRefVal = nextResolvedRef.value;
+                                if (externalRef) {
+                                    current[i] = resolvedRefVal;
+                                    break;
+                                }
+                            } else {
+                                resolvedRefVal = null;
+                                break;
+                            }
+                        }
+                    } else {
+                        queue.push(current[i]);
+                    }
+                }
+            } else if (typeof current === "object") {
+                for (const [key, value] of Object.entries(current)) {
+                    let resolvedRefVal = value;
+                    if (this.context.isReferenceObject(value)) {
+                        while (this.context.isReferenceObject(resolvedRefVal)) {
+                            const externalRef = this.context.isExternalReference(resolvedRefVal.$ref);
+                            const nextResolvedRef = await context.resolveReference({ $ref: resolvedRefVal.$ref });
+                            if (nextResolvedRef.resolved) {
+                                resolvedRefVal = nextResolvedRef.value;
+                                if (externalRef) {
+                                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                                    (current as any)[key] = resolvedRefVal;
+                                    break;
+                                }
+                            } else {
+                                resolvedRefVal = null;
+                                break;
+                            }
+                        }
+                    } else {
+                        queue.push(value);
+                    }
+                }
+            }
+        }
+        return spec;
+    }
```

Comment:
nit: this is a lot of nesting (is it possible to refactor into helper methods without more than ~3-4 levels of nesting for ease of readability